### ISSUE TITLE
AIP-74: Support grouping Asset by group attribute

### ISSUE THREAD

**Issue by Lee-W** [apache/airflow#43957]
### Description

<img width="688" alt="image" src="https://github.com/user-attachments/assets/bd1c052b-101c-4782-afeb-6f4a4b909411">

### Use case/motivation

https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-74+Introducing+Data+Assets

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Lee-W** [comment]
This will be picked up once most of the backend work of AIP-75 is done

**Comment by bbovenzi** [comment]
@Lee-W I'm wondering if we want to do this in our dependencies endpoint? Make a new asset group node type and add each asset as a dependency.

Or simpler, just add `asset_group` as a nullable field on the asset node model. Then the UI can infer the groups to make. Right now, we need an extra API call which would make the graph render twice.

**Comment by Lee-W** [comment]
> I'm wondering if we want to do this in our dependencies endpoint? Make a new asset group node type and add each asset as a dependency.

Sounds like a good idea 🤔 but might need some time to implement it

> Or simpler, just add asset_group as a nullable field on the asset node model. Then the UI can infer the groups to make. Right now, we need an extra API call which would make the graph render twice.

I thought we already have group info in the asset info?

**Comment by bbovenzi** [comment]
> Sounds like a good idea 🤔 but might need some time to implement it
> 
> I thought we already have group info in the asset info?

Yeah, this wont be for 3.0.0. We do have the group info when we request an Asset or a list of Assets. But not on the dependencies endpoint so its not ideal for rendering the graph.

**Comment by pierrejeambrun** [comment]
As there is no rush with this, I would suggest to do that in the dependency endpoint, so everything is done in the backend.

**Comment by bbovenzi** [comment]
@phanikumv no, this is more than an enhancement. We need to update how the dag dependencies are serialized. Marking at 3.1.0

**Comment by phanikumv** [comment]
Just checking in if  we have plans to get this done for 3.1.0?

**Comment by Lee-W** [comment]
I think this is missed. Would like to confirm what would be needed from the backend side again. Whether extending https://github.com/apache/airflow/blob/0204a89079123090ad6a792e4e373acfd7c19b70/airflow-core/src/airflow/models/serialized_dag.py#L654 to include asset group is sufficient for the frontend side?

**Comment by bbovenzi** [comment]
Not enough time for 3.1.0 moving to 3.2 where I think we'll be doing a lot more asset work.

I don't think the dag_dependencies response has any concept of grouping right now. So we would have to include the asset group and it would need to have `children`
